build-tooling.mk

```diff
@@ -144,37 +152,37 @@ publish:
 lint:
 ifneq ($(strip $(COMPONENT)),.)
 	cp .golangci.yaml $(COMPONENT)
-	$(DOCKER) build . -f Dockerfile --target lint --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
+	$(DOCKER) build . -f $(COMPONENT_DOCKERFILE) --target lint --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
 	rm -rf $(COMPONENT)/.golangci.yaml
 else
-	$(DOCKER) build . -f Dockerfile --target lint --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
+	$(DOCKER) build . -f $(COMPONENT_DOCKERFILE) --target lint --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
 endif
 
 .PHONY: fmt
 # Run go fmt against code
 fmt:
-	$(DOCKER) build . -f Dockerfile --target fmt --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
+	$(DOCKER) build . -f $(COMPONENT_DOCKERFILE) --target fmt --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
 
 .PHONY: vet
 # Perform static analysis of code
 vet:
-	$(DOCKER) build . -f Dockerfile --target vet --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
+	$(DOCKER) build . -f $(COMPONENT_DOCKERFILE) --target vet --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
 
 .PHONY: test
 # Run tests
 test: fmt vet
-	$(DOCKER) build . -f Dockerfile --target test --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
-	@$(DOCKER) build . -f Dockerfile --target unit-test-coverage --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY) --output build/$(COMPONENT)/coverage
+	$(DOCKER) build . -f $(COMPONENT_DOCKERFILE) --target test --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
+	@$(DOCKER) build . -f $(COMPONENT_DOCKERFILE) --target unit-test-coverage --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY) --output build/$(COMPONENT)/coverage
 
 .PHONY: binary-build
 # Build the binary
 binary-build:
-	$(DOCKER) build . -f Dockerfile --build-arg LD_FLAGS="$(LD_FLAGS)" --target bin --output build/$(COMPONENT)/bin --platform ${PLATFORM} --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
+	$(DOCKER) build . -f $(COMPONENT_DOCKERFILE) --build-arg LD_FLAGS="$(LD_FLAGS)" --target bin --output build/$(COMPONENT)/bin --platform ${PLATFORM} --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
 
 .PHONY: docker-build
 # Build docker image
 docker-build:
-	$(DOCKER) build . -t $(IMAGE) -f Dockerfile --target image --platform linux/amd64 --build-arg LD_FLAGS="$(LD_FLAGS)" --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
+	$(DOCKER) build . -t $(IMAGE) -f $(COMPONENT_DOCKERFILE) --target image --platform linux/amd64 --build-arg LD_FLAGS="$(LD_FLAGS)" --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
 
```

Comment:
Do we have a clean-up step where we remove the generated component dockerfiles?

Reply:
These files are added in dockerignore and gitignore. It can also be useful for debugging if something fails.

Reply:
i think we can have a flag or something to keep the intermediate dockerfiles for debugging purposes, but let's delete the intermediate dockerfiles after the build process to keep the project root directory clean

Reply:
Makes sense. I'll add a flag to toggle file deletions and delete by default.

Reply:
Added the `KEEP_COMPONENT_DOCKERFILE` arg (defaults to false). If set to true, retains the component Dockerfiles.

Reply:
Let's say we are running this in pipelines and want to debug issues with the generated dockerfiles, simply keeping them might not be enough. Even if we keep, how do we access it? Instead, can we print the generated dockerfiles for each component on the console?

Reply:
Makes sense, I think a better way would be to get these files as pipeline artifacts. Since, the flag can be toggled only if we configure the argument to true, which happens in pipeline configs. Thoughts ?

Reply:
Thanks! If there are ways to get the generated files as artifacts, then we are good. No need of printing in console.

Reply:
Resolving as these changes are not required now